# PGA

PGA is not an independent assembly software, it relies on the output of hifiasm and technologies that can anchor contigs to chromosomes,such as genetic maps, HIC, or alignment with genome sequences of related species to achieve a gapless genome.

# Dependencies

python3

mummer

networkx

# Installation

conda install -c bioconda mummer

conda install -c conda-forge networkx

git clone https://github.com/likui345/PGA.git

python setup.py install

# Usage

### 1.Using reference genomes to anchor scaffolds.


usage: 
    rbsa.py  --type nucmer --ref REF --scf SCF
    or rbsa.py  --type mcscan --A_bed A_BED --B_bed B_BED --anchor ANCHOR --scf SCF
    
optional arguments:
  -h, --help            show this help message and exit
  --type {nucmer,mcscan}
                        nucmer or mcscan anchor
  --ref REF             This is reference genome sequences which has anchored
                        to the chromosomal level
  --scf SCF             This is the scaffold sequences
  --A_bed A_BED         species_a bed file
  --B_bed B_BED         species_b bed file
  --anchor ANCHOR       anchor.simple files of species_a and species_b
    

### 2.Filtering overlap data .


usage: 
    filter.py  --paf_fn PAF --bestn NUM --output OUTPUT
    
optional arguments:
  -h, --help       show this help message and exit
  --paf_fn PAF_FN  Input. reads alignment file
  --bestn BESTN    output at least best n overlaps on 5' or 3' ends if
                   possible.
  --output OUTPUT  Output filename

### 3.Get chr_paths using agp file and gfa file.


usage: chr_paths.py  --agp AGP --gfa GFA

optional arguments:
  -h, --help  show this help message and exit
  --agp AGP   agp file form contig anchoring
  --gfa GFA   gfa from hifiasm

### 4.Building string graph.


usage: ovlp2graph.py [-h] --overlap-file OVERLAP_FILE

optional arguments:
  -h, --help            show this help message and exit
  --overlap-file OVERLAP_FILE
                        a file that contains the overlap information. (default: None)

Outputs:
    - sg_edges_list
    - chimer_nodes 

### 5.Generate the final assembly.


usage: graph2chr.py [-h] --reads-fasta-fn READS_FASTA_FN --paf-fn PAF_FN
                    --sg-edges-list-fn SG_EDGES_LIST_FN --ctg-paths-fn
                    CTG_PATHS_FN

optional arguments:
  -h, --help            show this help message and exit
  --reads-fasta-fn READS_FASTA_FN
                        Input. reads file generated by hifiasm. (default:
                        None)
  --paf-fn PAF_FN       Input. reads alignment file generated by hifiasm.
                        (default: None)
  --sg-edges-list-fn SG_EDGES_LIST_FN
                        Input. File containing string graph edges, produced by
                        ovlp2graph.py. (default: None)
  --ctg-paths-fn CTG_PATHS_FN
                        Input. File containing contig paths. (default: None)

